### PR TITLE
Ported everything to .NET Standard 1.6.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
This is a less ambitious and cut-down #175. Yet it has some thing missing:

- [x] The libraries are ported.
- [ ] The build script needs to be adjusted.

Edits from maintainers are allowed, so feel free to contribute.

### CODE REVIEW

**Comment by haf** [comment]
Thank you @teo-tsirpanis 

Two quick questions;

 - To restore and compile `dotnet restore && dotnet build`, right?
 - Assuming you have 1.0 of .Net Core installed?

**Comment by haf** [comment]
Having a look...

**Comment by haf** [comment]
Letting CI do its job...

**Comment by haf** [comment]
You'll have to tell me; does building for this netstandard1.6 support using the v2.0 toolchain and thereby a single `sln` file and single `fsproj` files and "magic paket"? /cc @matthid

**Comment by haf** [comment]
And should we have this restriction in paket.dependencies?

```
framework: net461, netstandard1.6, netcoreapp1.1
```

File: Expecto.FsCheck/Expecto.FsCheck.fsproj [hunk 1]
```diff
@@ -1,105 +1,15 @@
...
-    <Reference Include="mscorlib" />
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.Numerics" />
-    <ProjectReference Include="..\Expecto\Expecto.fsproj">
-      <Name>Expecto</Name>
-      <Project>{c0d55728-10a9-4a7a-9df9-d2f21f663ac2}</Project>
-      <Private>True</Private>
-    </ProjectReference>
+    <PackageReference Include="FSharp.NET.Sdk" Version="1.0.*" PrivateAssets="All" />
```
Comment (matthid) [inline Expecto.FsCheck/Expecto.FsCheck.fsproj]: This needs to be removed for the v2 toolchain to work.

File: Expecto.FsCheck/Expecto.FsCheck.fsproj [hunk 2]
```diff
@@ -1,105 +1,15 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+﻿<Project Sdk="FSharp.NET.Sdk;Microsoft.NET.Sdk">
```
Comment (matthid) [inline Expecto.FsCheck/Expecto.FsCheck.fsproj]: -FSharp.NET.Sdk

File: Expecto.Sample/Expecto.Sample.fsproj [hunk 3]
```diff
@@ -1,179 +1,20 @@
-﻿<?xml version="1.0" encoding="utf-8" standalone="yes"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project Sdk="FSharp.NET.Sdk;Microsoft.NET.Sdk">
```
Comment (matthid) [inline Expecto.Sample/Expecto.Sample.fsproj]: -FSharp.NET.Sdk

File: Expecto.Sample/Expecto.Sample.fsproj [hunk 4]
```diff
@@ -1,179 +1,20 @@
...
-    </ProjectReference>
-    <ProjectReference Include="../Expecto.BenchmarkDotNet/Expecto.BenchmarkDotNet.fsproj">
-      <Name>Expecto.BenchmarkDotNet.fsproj</Name>
-      <Project>{0acabeaa-29bd-4f9f-89db-0d03b5afc952}</Project>
-    </ProjectReference>
-    <ProjectReference Include="../Expecto.FsCheck/Expecto.FsCheck.fsproj">
-      <Name>Expecto.FsCheck.fsproj</Name>
-      <Project>{58507646-335a-460e-b283-77a2baf0d3a8}</Project>
-    </ProjectReference>
+    <PackageReference Include="FSharp.NET.Sdk" Version="1.0.*" PrivateAssets="All" />
```
Comment (matthid) [inline Expecto.Sample/Expecto.Sample.fsproj]: -FSharp.NET.Sdk

File: Expecto/Expecto.fsproj [hunk 5]
```diff
@@ -1,128 +1,21 @@
-<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project Sdk="FSharp.NET.Sdk;Microsoft.NET.Sdk">
```
Comment (matthid) [inline Expecto/Expecto.fsproj]: -FSharp.NET.Sdk

File: Expecto/Expecto.fsproj [hunk 6]
```diff
@@ -1,128 +1,21 @@
...
-          <HintPath>..\packages\System.ValueTuple\lib\netstandard1.0\System.ValueTuple.dll</HintPath>
-          <Private>True</Private>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
-</Project>
+  <ItemGroup>
+    <PackageReference Include="FSharp.NET.Sdk" Version="1.0.*" PrivateAssets="All" />
```
Comment (matthid) [inline Expecto/Expecto.fsproj]: -FSharp.NET.Sdk

**Comment by haf** [comment]
I'm sorry, but I don't get this to compile. What's the command you use to compile it?

**Comment by haf** [comment]
Still not compiling even after removing it.

**Comment by haf** [comment]
The magic is to rename `paket.bootstrapper.exe` to `paket.exe`, and place it in `.paket/`, having dotnet core invoke it from `paket.targets`.

**Comment by haf** [comment]
Ok, that's closer... Now there's:

```
/Users/h/dev/haf/expecto/Expecto.Tests/Expecto.Tests.fsproj : error NU1202: Package BenchmarkDotNet.Toolchains.Roslyn 0.10.9 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package BenchmarkDotNet.Toolchains.Roslyn 0.10.9 supports: net46 (.NETFramework,Version=v4.6) [/Users/h/dev/haf/expecto/Expecto.sln]
/Users/h/dev/haf/expecto/Expecto.BenchmarkDotNet/Expecto.BenchmarkDotNet.fsproj : error NU1202: Package BenchmarkDotNet.Toolchains.Roslyn 0.10.9 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package BenchmarkDotNet.Toolchains.Roslyn 0.10.9 supports: net46 (.NETFramework,Version=v4.6) [/Users/h/dev/haf/expecto/Expecto.sln]
/Users/h/dev/haf/expecto/Expecto.Sample/Expecto.Sample.fsproj : error NU1202: Package BenchmarkDotNet.Toolchains.Roslyn 0.10.9 is not compatible with netcoreapp1.1 (.NETCoreApp,Version=v1.1). Package BenchmarkDotNet.Toolchains.Roslyn 0.10.9 supports: net46 (.NETFramework,Version=v4.6) [/Users/h/dev/haf/expecto/Expecto.sln]
```

**Comment by haf** [comment]
Thanks for reviewing so far @matthid

**Comment by matthid** [comment]
Oh I think this might be quite an unfortunate situation, it seems like paket shouldn't write `BenchmarkDotNet.ToolChains.Roslyn` into the `netcoreapp1.1` references file (see `obj/` directory of the project) apparently we shouldn't do that because it only is a reference when building for net46. I guess this would take me some time to analyze and fix therefore it will take me a couple of days to get into this. For now I don't have a good workaround. 

Maybe @forki can take a look, I think this should already work as we apply framework restrictions in this case.

Can you upload the generated paket files in the `obj/` directory of the project and the lockfile + paket.references? (Maybe open a issue on Paket with this)

One "workaround" is to have multiple project files for full-net and netcore :/

**Comment by teo-tsirpanis** [comment]
> What's the command you use to compile it?

After fixing the Logary namespace, a `dotnet restore` and a `dotnet build -c Release` should work. However I am getting the same error about `BenchmarkDotNet.Toolchains.Roslyn`.

**Comment by haf** [comment]
@matthid 

 - [paket.lock.zip](https://github.com/haf/expecto/files/1290781/paket.lock.zip)
 - [obj.zip](https://github.com/haf/expecto/files/1290785/obj.zip)
 - [Gemfile.lock.zip](https://github.com/haf/expecto/files/1290786/Gemfile.lock.zip)

**Comment by haf** [comment]
Roger that. I'm out for the evening. This is real cooperation!

**Comment by haf** [comment]
I haven't had time to circle back to this PR; @matthid is paket fixed?

**Comment by matthid** [comment]
Yes I think it was https://github.com/fsprojects/Paket/issues/2759

**Comment by AnthonyLloyd** [comment]
hi @teo-tsirpanis, Can I get a status on this PR. I'm in the process of moving Expecto from rake to fake and considering .Net Standard 2.0 etc.

**Comment by teo-tsirpanis** [comment]
Hi, @AnthonyLloyd, that's great news! Me too had [tried to port it to FAKE](https://github.com/haf/expecto/pull/175) in the past. You can continue from [my fork](https://github.com/teo-tsirpanis/expecto).

**Comment by AnthonyLloyd** [comment]
@teo-tsirpanis Thanks, I will do. I'll steal your DotNetCli and NuGet fsx code for a start. I'll do this tonight. If you can give me any suggestions that you see.

One goal I would like is that we can clone and build in Visual Studio/Code without installing anything and possible without even running build.cmd. Will need to do something with Logary.Facade for this.

And help on porting the whole to just DotNetCli would be great.

**Comment by AnthonyLloyd** [comment]
@teo-tsirpanis I'm closing this as the codebase has moved on and include much of these changes. Thanks

**Pull request closed without merging**